src/main/scala/org/bitcoins/core/script/locktime/LockTimeInterpreter.scala

```diff
@@ -130,20 +131,19 @@ trait LockTimeInterpreter extends BitcoinSLogger {
     * @return if the given script number is valid or not
     */
   def checkSequence(program : ScriptProgram, nSequence : ScriptNumber) : Boolean = {
-    val inputIndex = program.txSignatureComponent.inputIndex
+    val inputIndex = program.txSignatureComponent.inputIndex.toInt
     logger.debug("inputIndex: " + inputIndex)
     val transaction = program.txSignatureComponent.transaction
-    val txToSequence : ScriptNumber = ScriptNumber(transaction.inputs(inputIndex).sequence)
+    val txToSequence : Int64 = Int64(transaction.inputs(inputIndex).sequence.underlying)
 
-    if (program.txSignatureComponent.transaction.version < 2) {
+    if (program.txSignatureComponent.transaction.version < UInt32(2)) {
       logger.error("Transaction version is too low for OP_CSV")
       return false
     }
 
-    val nLockTimeMask : Long = TransactionConstants.sequenceLockTimeTypeFlag | TransactionConstants.sequenceLockTimeMask
-    val txToSequenceMasked : ScriptNumber = txToSequence & ScriptNumber(nLockTimeMask)
-
-    val nSequenceMasked : ScriptNumber = nSequence & ScriptNumber(nLockTimeMask)
+    val nLockTimeMask : UInt32 = UInt32(TransactionConstants.sequenceLockTimeTypeFlag.underlying | TransactionConstants.sequenceLockTimeMask.underlying)
```

Comment:
`sequenceLockTimeTypeFlag` and `sequenceLockTimeMask` are both `UInt32` so we shouldn't have to call `underlying`. If `|` is not implemented on UInt32 go ahead and implement and write a specification inside of `UInt32Spec`.